source/extensions/io_socket/io_uring/io_uring_impl.cc

```diff
@@ -0,0 +1,127 @@
+#include "source/extensions/io_socket/io_uring/io_uring_impl.h"
```

Comment:
I know this PR is a draft, but I think my biggest piece of feedback is related to the complexity of the source changes which seem relatively subtle and the limited testing present so far.  I would consider starting with just this factory and writing comprehensive tests for the functionality it provides.  Then move to implementing just the write method from the IO socket interface.  Then the read side (or vice versa).  On a PR after that fill in some of the other relevant methods.

Reply:
Ok. I'll try to split this PR. @SaveTheRbtz suggested the `IoUringImpl` wrapper could be reused in block IO, so I'll submit it separately (as a part of `source/common/io`?).